### PR TITLE
Initial version for nxos_ssh get_optics

### PR DESCRIPTION
**Pull request by ExaneServerTeam**
Implements nxos_ssh get_optics function

* | json does not work, we have to parse the Terminal
* Extracts current / input / ouput metrics
* Also retrieves Gib data
* Maps Gbic type to Generic YANG types (Reference the Gbics We own so far)

### CODE REVIEW

**Comment by ExaneServerTeam** [comment]
Hello
You will find an implementation of Nexus get_optics. I tries to map the openconfig Yand format.
Sadly as stated, the json pipe did not work on my plateform.

Vendor Type is pretty useful but I do not know where to place it in the Json file (See 
conntector_type_map for samples). I you have any idea on this matter it would be welcome.

File: napalm/nxos_ssh/nxos_ssh.py [hunk 1]
```diff
@@ -1528,3 +1539,100 @@ def get_vlans(self):
...
+                            "min": 0.0,
+                            "max": 0.0,
+                        },
+                    },
+                }
+                port_detail["physical_channels"]["channel"].append(optic_states)
+
+            port_detail["state"] = state
+            optics_detail[port] = port_detail
+            # print(port_detail)
```
Comment (mirceaulinic) [inline napalm/nxos_ssh/nxos_ssh.py:1636]: Can you remove this line please?

Comment (ExaneServerTeam) [reply napalm/nxos_ssh/nxos_ssh.py]: Spurious print statements have been removed with the last commit.

Comment (ExaneServerTeam) [reply napalm/nxos_ssh/nxos_ssh.py]: Updated with a sample test

File: napalm/nxos_ssh/nxos_ssh.py [hunk 2]
```diff
@@ -1563,8 +1563,11 @@ def get_optics(self):
 
         for port_ts in port_ts_l:
             port = port_re.search(port_ts).group(1)
+            # No transceiver is present in those case
             if "transceiver is not present" in port_ts:
                 continue
+            if "transceiver is not applicable" in port_ts:
```
Comment (mirceaulinic) [inline napalm/nxos_ssh/nxos_ssh.py:1569]: Just a personal curiosity... what does "transceiver is not applicable" mean in this context? :thinking:

Comment (ExaneServerTeam) [reply napalm/nxos_ssh/nxos_ssh.py:1569]: @mirceaulinic  the *not applicable* state applies to the ports related to the HP C7000 Enclosure B22 switches.

**Comment by ExaneServerTeam** [comment]
@mirceaulinic  & @ktbyers: A welcome addition would be to have the cisco type of Gbic set in the Json / Dict. Sample are 1000base-LH, 1000base-SX, 10Gbase-LR ...
Do you have any idea where it could be set ? I have tried to find a place in the YANG Transceiver Tree but can't really find out a good place . May be in one of those two descriptions ?

```
+--rw physical-channels
          +--rw channel* [index]
             +--rw index     -> ../config/index
             +--rw config
             |  +--rw index?                 uint16
             |  +--rw description?           string
             |  +--rw tx-laser?              boolean
             |  +--rw target-output-power?   decimal64
             +--ro state
                +--ro index?                 uint16
                +--ro description?           string

```

**Pull request merged**